cts/lab/CTStests.py

```diff
@@ -1313,7 +1323,7 @@ def __call__(self, node):
         elif rsc.managed():
             return self.failure("%s was not recovered and is inactive" % self.rid)
 
-        return self.success()
+        return 0 # Anything but None is success
```

Comment:
I'd rather not use 0 to indicate success and None indicate failure, when both of them are falsy and neither of them has any intrinsic meaning in this context. Could we do either `CrmExit` codes or booleans as the return values here?

There's no need for `self.failure()` to return `None` in the first place either, since that's the default return value and there's no other possibility.

I'd rather call `self.failure()` and then return `False` or `CrmExit.<error>` in the case of error, and return `True` or `CrmExit.OK` in the case of success.

We don't necessarily have to do this now though. The return codes are kind of messy in more places than this (within the same class).

Reply:
Yeah this will have to wait for a bigger overhaul. The proper "Pythonic" approach would be to change the CTS lab to use exceptions rather than return values for errors throughout.